### ISSUE TITLE
linter: false positive on method named catch `(GlobalExceptionFilter.catch)`

### ISSUE THREAD

**Issue by antoinezanardi** [oxc-project/oxc#18750]
### What version of Oxlint are you using?

1.42.0

### What command did you run?

`oxlint --type-aware -c configs/oxlint/oxlint.config.jsonc`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
"promise/valid-params": "error",
"promise/prefer-await-to-then": "error",
}
```

## Problem

Oxlint flags GlobalExceptionFilter.catch(...) as if it's a Promise catch call, triggering prefer-await-to-then and valid-params rules, but the call is to a regular (non-async) instance method named catch. The rules should only apply when the catch is invoked on a Promise/thenable.

## What happens
 
- Running the linter reports prefer-await-to-then / valid-params violations for the line that calls globalExceptionFilter.catch(...).
- The test calls an instance method catch on a newly constructed GlobalExceptionFilter() object — not a Promise — so those rules are a false positive here.

## Relevant snippet (from repo)

```ts
globalExceptionFilter.catch(exception, localMocks.filters.globalException.host);
// ^----- This is reported as a Promise, but the prototype is 
// `public catch(exception: unknown, host: ArgumentsHost): void {`
```

## Why this is wrong

The rules currently detect the .catch call purely by name/shape and assume it's Promise.prototype.catch; they do not confirm the callee is actually a Promise or thenable. In this code the callee is an instance (class) method — the linter should skip Promises-only rules in that case.

**Comment by lhilgert9** [comment]
The same problem occurs when you create a schema in Zod that calls the .catch method. For example:
```ts
import { z } from "zod/mini";

const schema = z.catch(z.optional(z.string()), undefined);
//               ^--- `Promise.catch()` requires 1 argument, but received 2.
```

**Comment by camc314** [comment]
I reprouced both cases here:

https://github.com/camc314/oxlint-repros/tree/main/issue-18750

We have the same behaviour as the eslint rule - please file an issue upstream.
